### DISCUSSION TITLE
Question regarding compreFace

### DISCUSSION THREAD

**Discussion by leccyril** [exadel-inc/CompreFace#811] [General] [answered]
Hello, 

Thank you for your great tool

My first question is to ask a confirmation...

It is not possible to play with only on real time picture like iphone or samsung did  for face detection ?. In mean i want to be sure that someone will not come with a picture and make working application or detection just with picture usurpation... we want to use face detection in real time.

It is possible to store for example 1 area with several allowed persons..

we have several areas and each area are allowed not only for one person, but for a set of person ?

Thank you

**Comment by pospielov** [comment]
Hello,
I didn't get the question about the real-time picture and how it relates to phones. Could you describe it once again? 
If you have several areas and each area has its own set of people, there are two ways to implement it:
1. Create one Facial recognition service with all of the people. Then add the logic of defining where this person can access on the client side.
2. Create a Facial recognition service for each area and add only allowed people to them
Which one is better depends on your case, but I think both are good.

**Reply by leccyril** [reply]
ok second seems great, but for recognition, it sends in response the person most close to your picture, what to do if we just want to have response => YES face is in collection => access granted and response => NO face is not present in collection => access not granted we can not add some parameters to the recognition service to give us only present or not ? the closest person is not really something needed i think in most uses cases ?

And for example we don't need the UI interface because we want to manage it with our own front interface for Roles, services etc... we could just delete UI docker and directly use AP and admin docker container ? then what can be the case to Have multiple App and multiple services ( for example services for area in same building ?) and app for different companies or buildings ?

thank you for your great tools

**Reply by pospielov** [reply]
The facial recognition service doesn't answer the question if the person is present in the collection or not. It answers the question what is the similarity of unknown faces to the faces in the collection. The reason is that there is no threshold that can 100% guarantee that. There is always an error. So you need according to your case choose the threshold that satisfies your needs. Here is our documentation that explains how to do it:
https://github.com/exadel-inc/CompreFace/blob/master/docs/Face-Recognition-Similarity-Threshold.md

**Reply by leccyril** [reply]
yes i alredy read this, so the better solution is to use recognition service (with add subject or add example subjects what is the best? with several picture of each subjects) then in response we analyze the list of subject returned if there is one more than 0.98 in similarity so we can tell for example he is in collection.... this is the best solution to achieve it ?

and for use custom build [ custom build](https://github.com/exadel-inc/CompreFace/blob/master/custom-builds/README.md) what we have to do ?

facial recognition is working if i want to connect to my phone it will say OK , or NOT OK ^^

Thanks !

**Reply by pospielov** [reply]
> we could just delete UI docker and directly use AP and admin docker container ?

Yes, but make sure you opened the port on `compreface-api` container, so you can make requests. As `compreface-ui` container contains `nginx` that now routes your requests to `admin` and `api`

> then what can be the case to Have multiple App and multiple services ( for example services for area in same building ?) and app for different companies or buildings ?

Do you mean different instances of CompreFace? Yes, this will work. 

> and for use custom build [custom build](https://github.com/exadel-inc/CompreFace/blob/master/custom-builds/README.md) what we have to do ?

Custom build is just a separate `docker-compose.jml` file, so you just download it and start as usual CompreFace. But make sure you download it from `Releases`, the current version in `master` branch won't work on the last images from DockerHub.

**Reply by leccyril** [reply]
on the UI demo we can add several APP and in each APP several services see those printscreen [ for app ](https://snipboard.io/XmfJWb.jpg) and for services in each APP [for services ](https://snipboard.io/6zVP7x.jpg)

**Reply by leccyril** [reply]
I think what is possible to do also is for each area we add subject, then we made face vefirification with all subjects in the area collection, if no one is OK so access is not allowed... we can use this method if no lot of subjects are ine the collection... because at the end the recognition is making lot of verification no ?

**Reply by pospielov** [reply]
yes, recognition works the same as verification, but on all photos.
But if you need to compare lots of photos, it's a much more effective to use recognition than using verification.

**Comment by leccyril** [comment, marked as answer]
I mean for face detection on IPHONE you have to be in front of the phone, you can not take a picture and present the picture to the face recognition.

Imagine a face detection to be allowed to enter in area, someone go on facebook take a picture from one person allowed an present the picture to be allowed to enter....

**Reply by pospielov** [reply]
So you mean liveness detection (or fraud detection). No CompreFace doesn't support it. There is no simple way to implement **reliable** passive liveness detection with 2d images. 
So I would recommend implementing it with active liveness detection, our `landmarks` plugin should help you with it.

**Reply by leccyril** [reply]
yes liveness detection. great how to use it in our project ? need some other dependencies ? (we have nothing yet)

**Reply by pospielov** [reply]
I don't know open source solutions for liveness detection, similar to CompreFace
I usually recommend this article for understanding how to implement liveness detection:
https://aws.amazon.com/blogs/industries/liveness-detection-to-improve-fraud-prevention-in-financial-institutions-with-amazon-rekognition/
Of course, you can use CompreFace instead of AWS rekognition
